### DISCUSSION TITLE
New logs interface is available in public beta - User feedback

### DISCUSSION THREAD

**Discussion by gnoireaux** [CleverCloud/Community#39] [Console]
Très utile ! Un grand merci à l'équipe !

**Comment by gnoireaux** [comment]
Question:
https://github.com/CleverCloud/Community/discussions/38 : the announcement discussion has been _locked_.
But you _do_ ask for feedback :
![image](https://github.com/CleverCloud/Community/assets/73658/b18b526e-d091-4e01-b35a-618e4ab566d9)

Why lock the discussion if you want feedback ?

**Reply by davlgd** [reply]
There is a complete section, letting users to organise feedbacks by topics on the new logs stack, better than to put everything in a mega thread.

**Comment by plduhoux** [comment]
Vraiment un superbe outil ! Bravo à l'équipe !

Je regrette juste les bandeaux larges dans la console "Build succeeded" et "Successfully deployed", est il possible à minima de mettre ces lignes en couleur ?

**Reply by davlgd** [reply]
Dans le cas d'un build avec une instance dédiée ? Dans le fonctionnel actuel de la nouvelle interface, les deux instances sont l'une à la suite de l'autre, chacune apportant ses propres logs. On y trouve bien ces infos, par exemple sur une de mes apps j'ai: 

```
...
2024-04-17T12:35:07.249Z Build succeeded in 4 minutes and 43 seconds
...
2024-04-17T12:35:49.812Z Successfully deployed in 0 minutes and 27 seconds
```

Mais je comprends que l'intégration précédente rendait ces lignes plus visibles. Sur le sujet, une reflexion est en cours pour voir comment faire ressortir certains évènements sous la forme de blocs spécifiques, on a déjà un PoC sur le sujet. 

Mais on discutera avec l'équipe pour savoir si on peut traiter le sujet de ces deux lignes spécifiques directement ou si l'on doit attendre d'avancer de manière plus globale sur ce point.

**Reply by plduhoux** [reply]
Oui tout à fait, on retrouve bien ces lignes, mais elles sont un peu invisibles. Sans refaire le bandeau large d'avant, juste un traitement couleur serait parfait. 

J'utilise des runners gitlab et dans la log des jobs gitlab, ces lignes (venant de clever) apparaissent en couleur, je trouve ça vraiment pratique.

![Capture d’écran 2024-04-20 à 10 17 37](https://github.com/CleverCloud/Community/assets/687923/e1704e98-9980-45be-8358-a1db1b779b09)
![Capture d’écran 2024-04-20 à 10 17 50](https://github.com/CleverCloud/Community/assets/687923/f39bd49a-8608-4312-8ad7-ae79ab2ef574)

**Comment by cedk** [comment]
L'indicateur du nombre d'instance ne semble pas être correcte. 0 pour l'instance en cours et 1 pour une veille.
Sinon c'est agréable d'avoir les messages multi-lignes dans le bon ordre :+1:

**Reply by davlgd** [reply]
On a constaté un comportement du genre lorsqu'il y a des upscale/downscale, selon le numéro de l'instance qui est stoppée/lancée. C'est un sujet que l'on est en train de régler côté API (il n'est pas lié au composant en lui-même)

**Comment by jpouillard** [comment]
Très bonne évolution, juste une suggestion : quand je clique sur "Logs" j'arrive systématiquement sur le filtre des "7 derniers jours". Ne serait il pas mieux d'accéder aux logs "temps réel" plutot ?

**Reply by davlgd** [reply]
Les logs en temps réel s'affichent lorsqu'il y a un déploiement en cours. Le reste du temps, sur certaines applications il ne se passe pas grand chose et la zone est relativement vide, du coup on a préféré afficher la liste des déploiements disponibles pour l'exploration (souvent les clients vont dans les logs pour chercher une infos). Et on peut aller voir le live assez facilement.

Cela pourra changer avec l'arrivée d'une indexation des logs qui ouvrira pas mal de possibilités complémentaires, mais en l'état on a fait le choix de toujours renvoyer une page avec des informations exploitables à l'utilisateur. Et dans tous les cas, on ne pourra jamais deviner si l'utilisateur veut voir du live ou non lorsqu'il va dans la zone des logs, donc on on aura toujours un moment où le choix par défaut (même si on le mettait côté utilisateur) ne sera pas le bon :D

**Reply by 7ute** [reply]
Idem, sur une instance un peu "bavarde", le log de 7j donnes des dizaines de milliers de lignes.
Le réglage par défaut fait sens, mais une possibilité de configuration en plus (ie: "vue par défaut") serait fort pratique (dans les options de logs, `console/settings/app-logs-options`).

**Comment by tcollignon** [comment]
Salut,
C'est vraiment mieux super boulot ! 
Proposition : Quand je vois mes logs deployed XXX, en dessous j'ai 2 lignes avec j'imagine des noms générés d'instance, avec une pour build et une pour run => possible d'indiquer plus clairement laquelle est build et l'autre run ?

**Reply by davlgd** [reply]
Normalement l'info est remontée dans le menu de gauche avec un logo spécifique pour les instances de build. Mais nous avons remarqué que l'info n'est pas toujours accessible côté API, le sujet est en cours de correction de notre côté. Mais on a déjà évoqué le fait d'afficher certaines informations dans les logs sur l'instance, il faut que l'on tranche qui doit envoyer ces infos, mais je note de préciser la nature de l'instance.

**Comment by zeid0ne** [comment]
Pour l'interface précédente une fois sur deux j'avais du blanc à la place des logs.
Et c'était long à charger.

Maintenant c'est nickel !

**Reply by StevenLeRoux** [reply]
Top ! merci de ton retour 🤩

**Comment by fredericalix** [comment]
Super boulot !
C'est une sacrée belle évolution de l'affichage des logs 👍 
Concernant le filter, est-il possible de faire des recherches avec une expression régulière du genre « j'aimerais afficher les lignes contenant le mot-clé GET et un autre mot-clé ITEMS » ?

**Reply by davlgd** [reply]
Non pour le moment c'est un filtre assez simple, mais ça fait partie des évolutions prévues ;)

**Reply by cedk** [reply]
Un autre point qui serait bien pour les filtres c'est que quand l'entrée est multi-ligne, c'est que l'ensemble des lignes de celle-ci soit affichée.

**Reply by davlgd** [reply]
Disons que de manière plus générale on travaille à pouvoir afficher le contexte de la ligne trouvée. Mais selon les cas c'est plus ou moins pertinent. Le sujet UX est ouvert, on devrait proposer une première solution dans l'interface publique avant d'aller plus loin, pour collecter des retours.

**Comment by shulard** [comment]
Un sacré refresh de l'interface ! Les logs triés à la milliseconde c'est vraiment top, merci à l'équipe :+1:
